[Api docs]Added alert dialog dartpad demo

Alabhya268 opened the pull request:
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*


## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

Alabhya268 committed:
Added alert dialog demo

Alabhya268 committed:
Added alert dialog demo

Alabhya268 committed:
Added alert dialog demo

Alabhya268 committed:
Added alert dialog demo

Alabhya268 committed:
Added alert dialog demo

Alabhya268 committed:
Added alert dialog demo

Alabhya268 committed:
Updated branch

Alabhya268 commented:
@shihaohong ,Please review this PR.

Alabhya268 committed:
Resolved

Alabhya268 committed:
Resolved

Alabhya268 committed:
Resolved

Alabhya268 committed:
Resolved

Alabhya268 committed:
Resolved

Alabhya268 committed:
Resolved

Alabhya268 committed:
disabled formattor

Alabhya268 committed:
disabled formattor

Alabhya268 committed:
disabled formattor

Alabhya268 committed:
disabled formattor

Alabhya268 committed:
disabled formattor

Alabhya268 committed:
disabled formattor

Alabhya268 commented:
@shihaohong @Piinks ,Please review this PR.

Anas35 committed:
removed new keyword (#75049)

goderbauer committed:
Migrate driver tests in example/ to NNBD (#75022)

dnfield committed:
Reland integration_test NNBD (#75074)

* Revert "Revert "NNBD integration_test (#74922)" (#75030)"

This reverts commit 87b043658bc3eb16cf1032cbc9b0ba6484271cea.

* fix bad merge issues

goderbauer committed:
Migrate off e2e (#75029)

jmagman committed:
Revert "Migrate driver tests in example/ to NNBD (#75022)" (#75161)

This reverts commit 50a8ecb7d6cd1fe34bd7982f463909fb649e98e8.

Piinks reviewed (commented):
Hi @Alabhya268 Thanks for the contribution! Can you fill out the PR description template? Also, it looks like several checks are failing here.

Alabhya268 committed:
Update dialog.dart

Alabhya268 committed:
Update dialog.dart

Alabhya268 committed:
Update dialog.dart

Alabhya268 committed:
Update dialog.dart

Alabhya268 committed:
Update dialog.dart

Alabhya268 committed:
Update dialog.dart

Alabhya268 committed:
Added description

Alabhya268 committed:
Added description

Alabhya268 committed:
Added description

Alabhya268 committed:
Added description

Alabhya268 committed:
Added description

Alabhya268 committed:
Added description

Alabhya268 committed:
updated description

Alabhya268 committed:
updated description

Alabhya268 committed:
updated description

Alabhya268 committed:
updated description

Alabhya268 committed:
updated description

Alabhya268 committed:
updated description

Alabhya268 commented:
@Piinks , I've added the description.

chunhtai committed:
fixes preceision error during SliverGridRegularTileLayout (#74851)

jmagman committed:
Split tools_tests subshards into separate shards to support sub-sub-sharding (#75033)

creativecreatorormaybenot committed:
Fix formatting in rendering/box (#75102)

jmagman committed:
Avoid iOS local networking dialog in integration tests (#75163)

Commit:
Remove scrollability by default (#75138)

engine-flutter-autoroll committed:
Roll Engine from 833d9953a742 to 9a2795ae17ca (20 revisions) (#75140)

Piinks commented on packages/flutter/lib/src/material/dialog.dart:
```diff
@@ -218,6 +218,62 @@ class Dialog extends StatelessWidget {
+///     });
+///   }
+///
+///   return Scaffold(
+///     appBar: AppBar(
+///       title: Text('AlertDialog Demo'),
+///     ),
+///     body: Center(
```
The template already adds a Scaffold & Center widget. They are not needed here.

Piinks commented on packages/flutter/lib/src/material/dialog.dart:
````diff
@@ -218,6 +218,62 @@ class Dialog extends StatelessWidget {
+/// ```dart
+/// @override
+/// Widget build(BuildContext context) {
+///   Future<void> _showMyDialog() async {
````
Does this need to be in the build method?

Piinks replied:
Or, even at all? since `_showDialog` is just `=> showDialog`?

Piinks commented on packages/flutter/lib/src/material/dialog.dart:
```diff
@@ -218,6 +218,62 @@ class Dialog extends StatelessWidget {
+///       context: context,
+///       builder: (BuildContext context) => AlertDialog(
+///         title: const Text('AlertDialog tilte'),
+///         content: const Text(
+///           'AlertDialog description',
+///         ),
```
Nit: One line
```suggestion
///         content: const Text('AlertDialog description'),
```

Piinks commented on packages/flutter/lib/src/material/dialog.dart:
```diff
@@ -218,6 +218,62 @@ class Dialog extends StatelessWidget {
+///     return showDialog<String>(
+///       context: context,
+///       builder: (BuildContext context) => AlertDialog(
+///         title: const Text('AlertDialog tilte'),
```
Typo and capitalization.
```suggestion
///         title: const Text('AlertDialog Title'),
```

Piinks commented on packages/flutter/lib/src/material/dialog.dart:
```diff
@@ -218,6 +218,62 @@ class Dialog extends StatelessWidget {
 ///
+/// {@tool dartpad --template=stateful_widget_scaffold_center}
+///
+/// This demo shows a method in a [State] which, when called, displays a dialog box
+/// and returns a [Future] that completes when the dialog is dismissed.
```
This should at least mention the `showDialog` method.

Piinks commented on packages/flutter/lib/src/material/dialog.dart:
```diff
@@ -218,6 +218,62 @@ class Dialog extends StatelessWidget {
+///           ),
+///         ],
+///       ),
+///     ).then((returnVal) {
```
What is the type here?

Piinks commented on packages/flutter/lib/src/material/dialog.dart:
```diff
@@ -218,6 +218,62 @@ class Dialog extends StatelessWidget {
+///           SnackBar(
+///             duration: Duration(seconds: 2),
+///             content: Text('You clicked: $returnVal'),
+///             action: SnackBarAction(label: 'OK', onPressed: () {}),
```
A `SnackBarAction` that does not do anything isn't necessary.

Hixie committed:
Unpin mysteriously pinned packages (#74885)

robert-ancell committed:
Make Linux apps non-unique by default. (#74959)

This stops them attempting to own a D-Bus name, which may not be possible if the
app is sandboxed (e.g. in a Snap). It's also probably what most developers would
expect is the correct behaviour by default.

LongCatIsLooong committed:
Move caret/highlight painting to custom painters (#72828)

xu-baolin committed:
Reland "Update PopupMenuButton to match Material Design spec" (#74620)

jmagman committed:
Dump logs on failing devicelab test to recipe artifact location (#74378)

jmagman committed:
Replace context FakeUsage with TestUsage (#75170)

Hixie committed:
Remove the timeout when launching DevTools (#74859)

engine-flutter-autoroll committed:
Roll Engine from 9a2795ae17ca to aff8465afd80 (5 revisions) (#75174)

jlubeck committed:
Adds First/Last buttons to PaginatedDataTable (#73964)

HansMuller committed:
Added ButtonStyle.splashFactory and NoSplash.splashFactory (#74949)

jmagman committed:
Adopt subshard naming convention in build_tests (#75179)

jmagman committed:
Replace MockConfig with TestConfig (#75044)

engine-flutter-autoroll committed:
Roll Plugins from 04b0b4b13f59 to 31a8b5c3d5be (2 revisions) (#75177)

engine-flutter-autoroll committed:
Roll Engine from aff8465afd80 to e59457b81854 (4 revisions) (#75181)

engine-flutter-autoroll committed:
30721bed3 Automatically add platform labels (flutter/plugins#3487) (#75184)

engine-flutter-autoroll committed:
bb2bbde4d Roll Skia from fde740c7f26e to a6c692b8844d (7 revisions) (flutter/engine#24118) (#75186)

engine-flutter-autoroll committed:
a87497f13 Add plugin issue query to README (flutter/plugins#3493) (#75190)

YeungKC committed:
Improve Dropdown Menu (#74906)

srawlins committed:
Remove "unnecessary" imports. (#75198)

In each library where an import is removed, the library uses some elements
provided by the import, BUT there is another import which provides all of the
same elements, and at least one more which the library uses.

In this change, we remove the imports which can be simply removed in favor of
the other already present imports.

See https://github.com/dart-lang/sdk/issues/44569 for more information.

theniceboy committed:
Allow long press delay duration for LongPressDraggable to be adjustable (#74592)

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

Alabhya268 committed:
changes template

engine-flutter-autoroll committed:
Roll Engine from bb2bbde4d02b to 7334085fc1b7 (2 revisions) (#75199)

engine-flutter-autoroll committed:
a45557608 [camera] Fix example reference in camera's doc (flutter/plugins#3472) (#75211)

Alabhya268 committed:
Updated dialog

Alabhya268 committed:
Updated dialog

Alabhya268 committed:
Updated dialog

Alabhya268 committed:
Updated dialog

Alabhya268 committed:
Updated dialog

Alabhya268 committed:
Updated dialog

Alabhya268 commented:
@Piinks , I've now removed unnecessary snackbar and updated description ,Is it ok or anything else needs to be changed?

Alabhya268 committed:
Updated branch

Alabhya268 committed:
Updated branch

Alabhya268 committed:
Updated branch

Alabhya268 committed:
Updated branch

Alabhya268 committed:
Updated branch

Alabhya268 committed:
Updated branch

Alabhya268 committed:
Revert "Updated branch"

This reverts commit 860b664781023636718148d50097093fe78b644c.

Alabhya268 committed:
Revert "Updated branch"

This reverts commit 860b664781023636718148d50097093fe78b644c.

Alabhya268 committed:
Revert "Updated branch"

This reverts commit 860b664781023636718148d50097093fe78b644c.

Alabhya268 committed:
Revert "Updated branch"

This reverts commit 860b664781023636718148d50097093fe78b644c.

Alabhya268 committed:
Revert "Updated branch"

This reverts commit 860b664781023636718148d50097093fe78b644c.

Alabhya268 committed:
Revert "Updated branch"

This reverts commit 860b664781023636718148d50097093fe78b644c.

engine-flutter-autoroll committed:
Roll Engine from 7334085fc1b7 to 8a01ce59bf70 (3 revisions) (#75234)

Commit:
[flutter_tools] skip web renderer defines unless web target is picked (#75160)

Commit:
[flutter_tools] switch dart defines to base64 to avoid windows control characters (#75027)

= gets escaped into %3D which seems to be tripping up cmake on windows since % is a control character. Switch to base64 encoding, since this does not have % nor , in the output character set.

This change is not trivially cherry pickable, and isn't tested on windows aside from my local, manual tests due to the planned CI work not being complete yet.

Fixes #75017
Fixes #74705

mraleph committed:
Add support for producing unstripped AOT snapshots. (#75243)

yjbanov committed:
[web] benchmark and optimize defaultTargetPlatform (#75037)

goderbauer committed:
Make flutter_driver be usable from a null-safe app (#75175)

jmagman committed:
Adopt tool_test shard builders (#75171)

engine-flutter-autoroll committed:
e4c4a2f20 Roll Skia from f9a0f1a9be9e to 931cad1d74bb (11 revisions) (flutter/engine#24137) (#75250)

goderbauer committed:
Revert "Make flutter_driver be usable from a null-safe app (#75175)" (#75266)

This reverts commit 9fd5242e65a146d6029799a55cd07f8d28ad94a0.

Piinks commented on packages/flutter/lib/src/material/dialog.dart:
```diff
@@ -218,6 +218,39 @@ class Dialog extends StatelessWidget {
 ///
+/// {@tool dartpad --template=stateless_widget_scaffold_center}
+///
+/// This demo shows a [TextButton] which ontap calls [showDialog] method, when called,
```
```suggestion
/// This demo shows a [TextButton] which when pressed, calls [showDialog]. When called, this method
```

Piinks reviewed (approved):
This LGTM with just a tweak to the description! 
It also looks like you may need to update your branch again, master may have been broken when you updated last. Thanks!

Commit:
[flutter_tools] remove MockFeatureFlags (#75259)

jmagman committed:
Remove hard-coded subshard numbers from the test runner (#75189)

Commit:
[flutter_tools] remove mockito from flutter symbolize test (#75263)

Alabhya268 committed:
Update packages/flutter/lib/src/material/dialog.dart

Updated description

Co-authored-by: Kate Lovett <katelovett@google.com>

Alabhya268 committed:
Update packages/flutter/lib/src/material/dialog.dart

Updated description

Co-authored-by: Kate Lovett <katelovett@google.com>

Alabhya268 committed:
Update packages/flutter/lib/src/material/dialog.dart

Updated description

Co-authored-by: Kate Lovett <katelovett@google.com>

Alabhya268 committed:
Update packages/flutter/lib/src/material/dialog.dart

Updated description

Co-authored-by: Kate Lovett <katelovett@google.com>

Alabhya268 committed:
Update packages/flutter/lib/src/material/dialog.dart

Updated description

Co-authored-by: Kate Lovett <katelovett@google.com>

Alabhya268 committed:
Update packages/flutter/lib/src/material/dialog.dart

Updated description

Co-authored-by: Kate Lovett <katelovett@google.com>

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
updated branch

Alabhya268 committed:
Merge branch 'dialog' of https://github.com/Alabhya268/flutter into dialog

Alabhya268 committed:
Merge branch 'dialog' of https://github.com/Alabhya268/flutter into dialog

Alabhya268 committed:
Merge branch 'dialog' of https://github.com/Alabhya268/flutter into dialog

Alabhya268 committed:
Revert "Revert "Updated branch""

This reverts commit a3cc9f17c36d37ad1940751c1393c764a3a0a989.

Alabhya268 committed:
Merge branch 'new' into dialog

Alabhya268 commented:
@Piinks , I seems to be doing something wrong, Could you please tell me how to update branch with master manually,
I tried git rebase  and this happed.

I'll soon open new PR with same changes.
Thanks in advance.
